submissions/management/commands/analyze_uploaded_file_sizes.py

```diff
@@ -0,0 +1,157 @@
+            end_date.strftime(DATE_FORMAT)
+        )
+        print(arg_echo_str)
+        end_date = self.validate_input_dates(start_date, end_date)
+        submission_data = self.load_data(start_date, end_date)
+        self.print_report(submission_data)
+
+    def validate_input_dates(self, start_date, end_date):
+        """
+        Checks the validity of start_date and end_date
+        Checks that the end date is not before LAST_DATE, and if it is, overrides it.
+        """
+        if start_date < end_date:
+            raise CommandError("End date must be before start date")
+        if datetime.date.today() < start_date:
+            print(u"Warning: start_date is in the future")
+        if end_date < LAST_DATE:
+            print(u'File size data is unavailable before 2019-11-01. Setting end_date to 2019-11-01')
+            return LAST_DATE
+        return end_date
+
+    def load_data(self, start_date, end_date):
+        """ Load the uploaded file data from the database """
+        # The model fields are datetimes, we need to make sure we
+        # include everything that was submitted in that day
+        end_datetime = self.beginning_of_day(end_date)
+        start_datetime = self.end_of_day(start_date)
+
+        return Submission.objects.filter(
+            created_at__range=(end_datetime, start_datetime),
+            student_item__item_type='openassessment',
+            answer__contains='files_sizes":['
+        ).order_by(
+            'student_item__course_id',
+            'student_item__student_id'
+        ).select_related('student_item').values_list(
+            'student_item__course_id',
+            'student_item__student_id',
+            'answer',
+        )
+
+    def beginning_of_day(self, dt):
+        """ Given a datetime, return a datetime that represents the first second of that day """
+        return datetime.datetime.combine(dt, datetime.time.min)
+
+    def end_of_day(self, dt):
+        """ Given a datetime, return a datetime that represents the last second of that day """
+        return datetime.datetime.combine(dt, datetime.time.max)
+
+    def print_report(self, submission_data):
+        """ Crunch numbers, print results """
+        print(HEADER)
+        course_bytes = 0
+        users = set()
+        prev_course_id = None
+
+        for course_id, student_id, answer in submission_data:
+            answer = json.loads(answer)
+            if course_id != prev_course_id and prev_course_id is not None:
+                self.print_row(prev_course_id, users, course_bytes)
+                course_bytes = 0
+                users = set()
+            user_submission_bytes = sum(answer['files_sizes'])
+            if user_submission_bytes:
+                users.add(student_id)
+                course_bytes += user_submission_bytes
+            prev_course_id = course_id
+
+        if prev_course_id:
+            self.print_row(prev_course_id, users, course_bytes)
```

Comment:
Or maybe turn this whole section into a helper function that returns lists of aggregated, per-course data, and let `print_report()` be in charge of only calling that function and printing the report?